docs/architecture/blazor-for-web-forms-developers/components.md

```diff
@@ -3,11 +3,531 @@ title: Build reusable UI components with Blazor
+
+```razor
+<input type="checkbox" @bind="isChecked" />
+
+@code {
+    bool isChecked;
+}
+```
+
+When the component is rendered, the value of the checkbox will be set to the value of the `isChecked` field. When the user toggles the checkbox, the `onchange` is fired and the `isChecked` field is set to the new value. This means that the `@bind` syntax in this case is equivalent to the following.
+
+```razor
+<input value="@isChecked" @onchange="(UIChangeEventArgs e) => isChecked = e.Value" />
+```
+
+To change the event used for the bind, use the `@bind:event` attribute.
+
+```razor
+<input @bind="text" @bind:event="oninput" />
+<p>@text</p>
+
+@code {
+    string text;
+}
+```
+
+Components can also support data binding to their parameters by defining an event callback parameter with the same name as the bindable parameter, but with the "Changed" suffix added.
+
+*PasswordBox.razor*
+
+```razor
+Password: <input 
+    value="@Password" 
+    @oninput="OnPasswordChanged" 
+    type="@(showPassword ? "text" : "password")" />
+
+<label><input type="checkbox" @bind="showPassword" />Show password</label>
+
+@code {
+    private bool showPassword;
+
+    [Parameter]
+    public string Password { get; set; }
+
+    [Parameter]
+    public EventCallback<string> PasswordChanged { get; set; }
+
+    private Task OnPasswordChanged(ChangeEventArgs e)
+    {
+        Password = e.Value.ToString();
+        return PasswordChanged.InvokeAsync(Password);
+    }
+}
+```
+
+Note that to chain a data binding to an underlying UI element you'll need to set the value and handle the event directly on the UI element instead of using the `@bind` attribute.
+
+To bind to a component parameter, use a `@bind-{Parameter}` attribute to specify which parameter you want to bind to.
+
+```razor
+<PasswordBox @bind-Password="password" />
+
+@code {
+    string password;
+}
+```
+
+## Component lifecycle
+
+ASP.NET Web Forms has well-defined lifecycle methods for modules, pages, and controls. Blazor components also have a well defined lifecycle that can be used to initialize component state and implement advanced component behaviors. 
+
+All of Blazor component lifecycle methods have both sync and async versions. This is important, because component rendering is synchronous. You cannot run async logic as part of the component rendering. All async logic must happen as part of an async lifecycle method.
+
+### OnInitialized
+
+The `OnInitialized` and `OnInitializedAsync` methods are used to initialize the component. A component is typically initialized after it is first rendered. After a component is initialized it may be rendered multiple types before it is eventually disposed. The `OnInitialized` method is very similar to the `Page_Load` event in ASP.NET Web Forms pages and controls.
```

Comment:
```suggestion
The `OnInitialized` and `OnInitializedAsync` methods are used to initialize the component. A component is typically initialized after it is first rendered. After a component is initialized, it may be rendered multiple types before it is eventually disposed. The `OnInitialized` method is very similar to the `Page_Load` event in ASP.NET Web Forms pages and controls.
```